frontend/src/scenes/insights/ActionFilter/ActionFilter.stories.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react'
+import { ActionFilter, ActionFilterProps } from './ActionFilter'
+import { personPropertiesModel } from '~/models/personPropertiesModel'
+import { cohortsModel } from '~/models/cohortsModel'
+import { TaxonomicFilterGroupType } from 'lib/components/TaxonomicFilter/types'
+import { taxonomicFilterMocksDecorator } from 'lib/components/TaxonomicFilter/__mocks__/taxonomicFilterMocksDecorator'
+import { useMountedLogic, useValues } from 'kea'
+import { ChartDisplayType, FilterType, InsightType } from '~/types'
+import { MathAvailability } from './ActionFilterRow/ActionFilterRow'
+import { groupsModel } from '~/models/groupsModel'
+import { alphabet } from 'lib/utils'
+import { ComponentStory } from '@storybook/react'
+
+export default {
+    title: 'Filters/Action Filter',
+    decorators: [taxonomicFilterMocksDecorator],
+}
+
+const Template: ComponentStory<typeof ActionFilter> = ({ ...props }: Partial<ActionFilterProps>) => {
+    useMountedLogic(personPropertiesModel)
+    useMountedLogic(cohortsModel)
+    const { groupsTaxonomicTypes } = useValues(groupsModel)
+
+    const [filters, setFilters] = useState<FilterType>({
+        insight: InsightType.TRENDS,
+        events: [
+            {
+                id: '$pageview',
+                name: '$pageview',
+                order: 0,
+                type: 'events',
+            },
+        ],
+    })
+
+    return (
+        <ActionFilter
+            filters={filters}
+            setFilters={(payload: Partial<FilterType>): void => setFilters(payload)}
+            typeKey={`trends_${InsightType.TRENDS}`}
+            buttonCopy="Add graph series"
+            buttonType="link"
+            showSeriesIndicator
+            entitiesLimit={
+                filters.insight === InsightType.LIFECYCLE || filters.display === ChartDisplayType.WorldMap
+                    ? 1
+                    : alphabet.length
+            }
+            mathAvailability={
+                filters.insight === InsightType.LIFECYCLE
+                    ? MathAvailability.None
+                    : filters.insight === InsightType.STICKINESS
+                    ? MathAvailability.ActorsOnly
+                    : MathAvailability.All
+            }
+            propertiesTaxonomicGroupTypes={[
+                TaxonomicFilterGroupType.EventProperties,
+                TaxonomicFilterGroupType.PersonProperties,
+                ...groupsTaxonomicTypes,
+                TaxonomicFilterGroupType.Cohorts,
+                TaxonomicFilterGroupType.Elements,
+            ]}
+            customRowPrefix={undefined}
+            {...props}
+        />
+    )
+}
+
+export const Standard = Template.bind({})
+Standard.args = {}
+
+export const FullWidth = Template.bind({})
+FullWidth.args = {
+    fullWidth: true,
+}
+
+export const HorizontalUI = Template.bind({})
+HorizontalUI.args = {
+    horizontalUI: true,
+}
```

Comment:
Something isn't hooked up right here: opening a filter in one story opens all of them.
![2022-05-11 12 25 50](https://user-images.githubusercontent.com/4550621/167829315-6ffb3760-20b5-44eb-a668-88dbff318316.gif)

Reply:
I'm not certain whats going on here either. It seems to be something about the Kea logic being shared perhaps. @mariusandra do you have some pointers here? My gut tells me that the components should be wrapped in some sort of Bind to keep them separate?